rpc/rpcserver/server.go

```diff
@@ -0,0 +1,678 @@
+package rpcserver
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+
+	"github.com/btcsuite/btcd/blockchain"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcutil"
+	"github.com/btcsuite/btcwallet/chain"
+	"github.com/btcsuite/btcwallet/internal/cfgutil"
+	"github.com/btcsuite/btcwallet/internal/zero"
+	"github.com/btcsuite/btcwallet/netparams"
+	pb "github.com/btcsuite/btcwallet/rpc/walletrpc"
+	"github.com/btcsuite/btcwallet/waddrmgr"
+	"github.com/btcsuite/btcwallet/wallet"
+)
+
+// translateError attempts to create a new gRPC error with an appropiate error
+// code for recognized errors.  If this can not be done, the original error is
+// returned.
+//
+// This function is by no means complete and should be expanded based on other
+// known errors.  Any RPC handler not returning a gRPC error (with grpc.Errorf)
+// should return this result instead.
+func translateError(err error) error {
+	switch {
+	case waddrmgr.IsError(err, waddrmgr.ErrWrongPassphrase):
+		return grpc.Errorf(codes.InvalidArgument, "%s", err.Error())
+	default:
+		return err
+	}
+}
+
+// Server provides wallet services for RPC clients.
+type Server struct {
+	wallet *wallet.Wallet
+}
+
+// LoaderServer provides RPC clients with the ability to load and close wallets,
+// as well as establishing a RPC connection to a btcd consensus server.
+type LoaderServer struct {
+	loader    *wallet.Loader
+	activeNet *netparams.Params
+	rpcClient *chain.RPCClient
+	mu        sync.Mutex
+}
+
+var (
+	_ pb.WalletServiceServer       = (*Server)(nil)
+	_ pb.WalletLoaderServiceServer = (*LoaderServer)(nil)
+)
+
+func NewServer(wallet *wallet.Wallet) *Server {
+	return &Server{wallet}
+}
+
+func (s *Server) Ping(cxt context.Context, req *pb.PingRequest) (*pb.PingResponse, error) {
```

Comment:
Shouldn't all these `cxt` be `ctx`?  First, `ctx` matches the name used in `api.pb.go` and second, context is almost always abbreviated as `ctx`.